tests/pipeline/dataflowanalysis/test_expression_propagation_mem.py

```diff
@@ -22,6 +22,95 @@
 int32 = Integer.int32_t()
 int64 = Integer.int64_t()
 
+def test_address_propagation_does_not_break_relations_between_aliased_versions():
+    """
+        +------------------+
+        |        0.        |
+        |    x#0 = 0x0     |
+        |    y#0 = 0x0     | <--- DO NOT propagate
+        | ptr_x#1 = &(x#0) | <--- can propagate
+        | ptr_y#1 = &(y#0) | <--- can propagate
+        |  func(ptr_x#1)   |
+        |    y#1 = y#0     | <--- propagation will cause connection loss between lhs and rhs variable
+        |    x#1 -> x#0    |
+        |  func(ptr_y#1)   |
+        |    y#2 -> y#1    |
+        |    x#2 = x#1     |
+        |    x#3 = x#2     | <--- can propagate, cause x#2 is not used anywhere else
```

Comment:
Is it because x#3 or x#2 is not used. I thought x#3, because Dead code elimination can no remove this instruction, but I am not sure